pkg/scalers/neo4j_scaler.go

```diff
@@ -0,0 +1,199 @@
+	if s.driver != nil {
+		err := s.driver.Close(ctx)
+		if err != nil {
+			s.logger.Error(err, fmt.Sprintf("failed to close neo4j connection, because of %v", err))
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *neo4jScaler) getQueryResult(ctx context.Context) (float64, error) {
+	session := s.driver.NewSession(ctx, neo4j.SessionConfig{
+		AccessMode: neo4j.AccessModeRead,
+	})
+	defer session.Close(ctx)
+	result, err := session.ExecuteRead(ctx, matchItemFn(s.metadata.query, ctx))
+	if err != nil {
+		s.logger.Error(err, fmt.Sprintf("Couldn't execute query string because of %v", err))
+		return 0, err
+	}
+	res, err := strconv.ParseFloat((fmt.Sprintf("%v", result)), 64)
+	if err != nil {
+		s.logger.Error(err, fmt.Sprintf("Couldn't parse to int because of %v", err))
+		return 0, err
+	}
+	return res, nil
+}
+
+func matchItemFn(query string, ctx context.Context) neo4j.ManagedTransactionWork {
+	return func(tx neo4j.ManagedTransaction) (any, error) {
+		records, err := tx.Run(ctx, query, nil)
+		if err != nil {
+			return nil, err
+		}
+		if records.Peek(ctx) {
+			record := records.Record()
+			if err != nil {
+				return nil, err
+			}
+			return record.Values[1], nil
+		}
+
+		return 0, nil
+	}
+}
+
+func parseNeo4jMetadata(config *ScalerConfig) (*neo4jMetadata, string, error) {
+	var connStr string
+	meta := neo4jMetadata{}
+
+	if val, ok := config.TriggerMetadata["query"]; ok {
+		meta.query = val
+	} else {
+		return nil, "", fmt.Errorf("no query given")
+	}
+
+	if val, ok := config.TriggerMetadata["queryValue"]; ok {
+		queryValue, err := strconv.ParseFloat(val, 64)
+		if err != nil {
+			return nil, "", fmt.Errorf("failed to convert %v to int, because of %w", val, err)
+		}
+		meta.queryValue = queryValue
+	} else {
+		return nil, "", fmt.Errorf("no queryValue given")
+	}
+
+	meta.activationQueryValue = 0
+	if val, ok := config.TriggerMetadata["activationQueryValue"]; ok {
+		activationQueryValue, err := strconv.ParseFloat(val, 64)
+		if err != nil {
+			return nil, "", fmt.Errorf("failed to convert %v to int, because of %w", val, err)
+		}
+		meta.activationQueryValue = activationQueryValue
+	}
+
+	// Build connection str
+	host, err := GetFromAuthOrMeta(config, "host")
+	if err != nil {
+		return nil, "", err
+	}
+	meta.host = host
+
+	port, err := GetFromAuthOrMeta(config, "port")
+	if err != nil {
+		return nil, "", err
+	}
+	meta.port = port
+
+	username, err := GetFromAuthOrMeta(config, "username")
+	if err != nil {
+		return nil, "", err
+	}
+	meta.username = username
+
+	protocol, err := GetFromAuthOrMeta(config, "protocol")
+	if err != nil {
+		return nil, "", err
+	}
+	meta.protocol = protocol
+
+	if config.AuthParams["password"] != "" {
+		meta.password = config.AuthParams["password"]
+	} else if config.TriggerMetadata["passwordFromEnv"] != "" {
+		meta.password = config.ResolvedEnv[config.TriggerMetadata["passwordFromEnv"]]
+	}
+	if len(meta.password) == 0 {
+		return nil, "", fmt.Errorf("no password given")
+	}
+	
```

Comment:
Remove one of these 2 empty lines please

Reply:
Done